tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature

```diff
@@ -529,30 +529,114 @@ Feature: Sharing files and folders with internal users
     When the user disables all the custom permissions of collaborator "User One" for file "lorem.txt" using the webUI
     Then no custom permissions should be set for collaborator "User One" for file "lorem.txt" on the webUI
     And user "user1" should have received a share with these details:
-      | field       | value               |
-      | uid_owner   | user2               |
-      | share_with  | user1               |
-      | file_target | /lorem (2).txt      |
-      | item_type   | file                |
-      | permissions | read                |
+      | field       | value          |
+      | uid_owner   | user2          |
+      | share_with  | user1          |
+      | file_target | /lorem (2).txt |
+      | item_type   | file           |
+      | permissions | read           |
 
   Scenario Outline: share a file with another internal user assigning a role and the permissions
     Given user "user2" has logged in using the webUI
     When the user shares file "lorem.txt" with user "User One" as "<role>" with permissions "<collaborators-permissions>" using the webUI
     Then user "User One" should be listed as "<displayed-role>" in the collaborators list for file "lorem.txt" on the webUI
     And custom permissions "<displayed-permissions>" should be set for user "User One" for file "lorem.txt" on the webUI
+    And user "user1" should have received a share with these details:
+      | field       | value          |
+      | uid_owner   | user2          |
+      | share_with  | user1          |
+      | file_target | /lorem (2).txt |
+      | item_type   | file           |
+      | permissions | <permissions>  |
+    Examples:
+      | role        | displayed-role | collaborators-permissions | displayed-permissions | permissions         |
+      | Viewer      | Viewer         | share                     | share                 | read, share         |
+      | Editor      | Editor         | share                     | share                 | read, share, change |
+      | Custom Role | Editor         | share, change             | share                 | read, share, change |
+
+  Scenario Outline: share a folder with another user with share permissions and reshare with different user
+    Given user "user3" has been created with default attributes
+    And user "user2" has logged in using the webUI
+    When the user shares folder "simple-folder" with user "User One" as "<role>" with permissions "<collaborators-permissions>" using the webUI
```

Comment:
we know that sharing works correctly by running the previous tests, lets do this steps in API and test the resharing only